get_coverage_for_challenge.sh

```
@@ -53,17 +53,20 @@ mkdir -p ${CSHARP_TEST_COVERAGE_DIR}
 )
 
 # Run the tests against the instrumented binaries
+FULL_PATH_TO_NUNIT_CONSOLE="$(cd ${SCRIPT_CURRENT_DIR} && find . -path *nunit*console.exe | head -n 1 || true)"
+
 (
   cd ${SCRIPT_CURRENT_DIR} && \
     mono ${SCRIPT_CURRENT_DIR}/packages/altcover.3.5.569/tools/net45/AltCover.exe Runner                \
```

We can't have the user change ALTCOVER - so if we can't find this - should we throw an error or message during build?

---

If the definition is removed from the project, the same thing should appear and we could return a 0 as coverage and a message that the coverage tool has been removed - at least for the .net projects since we don't have many alternatives to do code coverage.

Let me know your thoughts and I will come up with a PR

---

This looks good. It should handle changes to the nunit library

Regarding other missing dependencies, we should not return 0 as the coverage as this is a valid, normal output and we would not pick this up as an error. 
In the case of the current issue, the report was sent to the client with 0 coverage and it was misleading. The correct behaviour is to throw an error and return a non-zero status code.
This will be then picked up buy the AWS monitoring and hopefully will be spotted by me before getting in front of the client (dpnt-infra-events)

---

Yes, this is a change in behaviour. 
It will make the scripts easier to support in Prod.